qa/valgrind.supp

```diff
@@ -1,3 +1,11 @@
+
+{
+   <allthefrees, so we can behave with tcmalloc>
+   Memcheck:Free
+   fun:free
+   ...
+}
```

Comment:
this `MismatchedFree` report looks new in the rgw suite:
```
<error>
  <unique>0xeef7eb</unique>
  <tid>1</tid>
  <kind>MismatchedFree</kind>
  <what>Mismatched free() / delete / delete []</what>
  <stack>
    <frame>
      <ip>0x4C32EA0</ip>
      <obj>/usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so</obj>
      <fn>operator delete[](void*, unsigned long)</fn>
      <dir>/builddir/build/BUILD/valgrind-3.16.0/coregrind/m_replacemalloc</dir>
      <file>vg_replace_malloc.c</file>
      <line>660</line>
    </frame>
    <frame>
      <ip>0x5FBEB86</ip>
      <obj>/usr/lib64/librados.so.2.0.0</obj>
    </frame>
    <frame>
      <ip>0x5FBEF2C</ip>
      <obj>/usr/lib64/librados.so.2.0.0</obj>
    </frame>
    <frame>
      <ip>0x5F7F7C5</ip>
      <obj>/usr/lib64/librados.so.2.0.0</obj>
      <fn>librados::v14_2_0::Rados::shutdown()</fn>
    </frame>
```
are we missing something in the suppression, or is this unrelated to tcmalloc?

Reply:
hmm, not quite sure why it's different, but i only tested the new suppression+ normal build with rados/verify --filter valgrind and rados/valgrind-leaks (which ensure we correctly detect a deliberate leak).  those test cases don't use librados.  also.. librados wouldn't use tcmalloc unless the application using it uses tcmalloc. maybe that changes the path somehow?

in any case, it should be easy to test this locally and see if it is related to -DALLOCATOR=libc vs tcmalloc and, if so, add another suppression.